book/tests/end_to_end.rs

```diff
@@ -0,0 +1,174 @@
+use tarpc::context;
+
+#[macro_use]
+extern crate diesel_migrations;
+
+use book::models::Error;
+
+const DATABASE_URL: &str = "postgres://postgres:password@localhost";
+const RPC_SERVER_ADDR: &str = "127.0.0.1:0"; // get an ephemeral address
+
+#[tokio::test]
+async fn should_not_find_book() {
+    // Arrange
+    let (_, mut client) = utils::test_setup().await;
+
+    // Act
+    let book_not_found = client.get_book(context::current(), 4).await.unwrap();
+
+    // Assert
+    assert_eq!(
+        book_not_found.unwrap_err(),
+        Error::NotFound,
+        "Database should be empty"
+    );
+}
+
+#[tokio::test]
+async fn should_find_book() {
+    // Arrange
+    let (_, mut client) = utils::test_setup().await;
+
+    // Act
+    let book_not_found = client.get_book(context::current(), 1).await.unwrap();
+
+    // Assert
+    assert_eq!(
+        book_not_found.unwrap(),
+        test_data::book_1(),
+        "Database should be empty"
+    );
+}
+
+mod test_data {
+    use chrono::NaiveDate;
+
+    use book::models::*;
+
+    pub fn book_1() -> Book {
+        Book {
+            id: 1,
+            code_identifier: "SE001".to_string(),
+            isbn: "1234567891011".to_string(),
+            issn: None,
+            release_date: NaiveDate::from_ymd(1900, 1, 1),
+            subtitle: Some("Why you should learn it".to_string()),
+            title: "Rust Book".to_string(),
+            category: Category {
+                id: 1,
+                name: "STS".to_string(),
+            },
+            language: Language {
+                id: 1,
+                iso_code: "aar".to_string(),
+                name: "Afar".to_string(),
+            },
+            publisher: Publisher {
+                id: 1,
+                name: "Nostarch Press".to_string(),
+            },
+            series: None,
+            authors: vec![
+                Person {
+                    id: 1,
+                    first_name: "Markus".to_string(),
+                    last_name: "Merklinger".to_string(),
+                    isni: None,
+                    orcid: None,
+                    oclc: None,
+                },
+                Person {
+                    id: 2,
+                    first_name: "Maya".to_string(),
+                    last_name: "Alroy".to_string(),
+                    isni: None,
+                    orcid: None,
+                    oclc: None,
+                },
```

Comment:
Thanks for the honour. Though, I would prefer to not be part of the tests. :-D

Reply:
Okay :)